Source/WTF/wtf/text/WTFString.h

```diff
@@ -58,14 +58,16 @@ class String final {
 
     // Construct a string with UTF-16 data.
     WTF_EXPORT_PRIVATE String(const UChar* characters, unsigned length);
-    ALWAYS_INLINE String(Span<const UChar> characters) : String(characters.data(), characters.size()) { }
 
     // Construct a string with Latin-1 data.
     WTF_EXPORT_PRIVATE String(const LChar* characters, unsigned length);
     WTF_EXPORT_PRIVATE String(const char* characters, unsigned length);
-    ALWAYS_INLINE String(Span<const LChar> characters) : String(characters.data(), characters.size()) { }
     ALWAYS_INLINE static String fromLatin1(const char* characters) { return String { characters }; }
 
+    // Construct a string from a span object.
+    ALWAYS_INLINE static String fromSpan(Span<const LChar> span) { return String { span.data(), static_cast<unsigned>(span.size()) }; }
```

Comment:
This should be kept grouped in with the Latin-1 constructors and should have Latin1 in its name, since unlike `UChar`, the type `LChar` is not strong enough to firmly establish the character set.

Reply:
With `fromLatin1Span(Span<const LChar>)`, should UChar variant be `fromUnicodeSpan(Span<const UChar>)`?

Reply:
I don't think so. When the type expresses things unambiguously I think it's good to not make the names so long and explicit. I think `fromSpan` is better.

Reply:
One downside to separating these two in name is not being able to use them as overloaded functions in templates where the character type is the template parameter. `ArgumentCoder<String>::decode()` and its `decodeStringText` helper are one current example.

It would be possible to constexpr-evaluate the character type and use the appropriate named constructor, but it's not necessarily pretty.

Reply:
This is not a problem. The encoder is switching to use `characters8()` or `characters16()`. It's natural that the decoder is also switching.
https://github.com/WebKit/WebKit/blob/4f6caeb7a6616e3ab6f4ea54a9ce483ad535386e/Source/WebKit/Platform/IPC/ArgumentCoders.cpp#L92-L95

Reply:
Comparative point of 8-bit branching is where the different `decodeStringText` specialization is selected. In this case reworking that codepath and removing the `decodeStringText` template altogether will probably keep things simple enough.